stustapay/core/service/customer.py

```diff
@@ -237,3 +273,90 @@ async def get_public_customer_api_config(self) -> PublicCustomerApiConfig:
             contact_email=self.cfg.customer_portal.contact_email,
             about_page_url=self.cfg.customer_portal.about_page_url,
         )
+
+    @with_db_transaction
+    @requires_customer
+    async def create_checkout(self, *, conn: asyncpg.Connection, current_customer: Customer, 
+                              amount: float) -> str:
+        # check amount
+        if amount <= 0:
+            raise InvalidArgument("Amount must be greater than 0")
+
+        amount = round(amount, 2)
+
+        # create checkout reference as uuid
+        checkout_reference = uuid.uuid4()
+        # check if checkout reference already exists
+        while await conn.fetchval("select exists(select * from checkout where checkout_reference = $1)", checkout_reference):
+            checkout_reference = uuid.uuid4()
+        
+
+        create_checkout = SumupCreateCheckout(
+            checkout_reference=checkout_reference,
+            amount=amount,
+            currency=self.config_service.get_public_config().currency_identifier,
+            merchant_code="",
+            pay_to_email="",
+            description="",
+            return_url="",
+            customer_id=f"{current_customer.user_tag_uid_hex}",
+            redirect_url="",
+        )
+
+        headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json',
+            'Authorization': f'Bearer {self.cfg.customer_portal.sumup_client_secret}'
+        }
+
+        # response = requests.request("POST", self.SUMUP_URL, headers=headers, data=create_checkout.json())
+
+        async with aiohttp.ClientSession(headers=headers) as session:
+            try:
+                async with session.post(self.SUMUP_URL, data=create_checkout.json(), timeout=2) as response:
+
+                    if not response.ok:
+                        logging.error(f"Sumup API returned status code {response.status} with body {response.text}")
+                        raise ServiceException("Sumup API returned an error")
+
+
+                    response_text = await response.text()
```

Comment:
shouldn't this be `response.json()` if are going to parse it into a pydantic object later on